### ISSUE TITLE
[FEATURE] Add docs for the Website

### ISSUE THREAD

**Issue by SaptarshiSarkar12** [SaptarshiSarkar12/Drifty#273]
### Is your feature request related to a problem? Please describe.

The README file currently lacks examples and use cases for both Drifty CLI and GUI. It also does not mention the custom issue labels that the project uses. 
Before testing their changes, the contributors need to run some commands. The addition of Dockerfiles to this project via #218 will require some notes and advices, but the README file is already too crowded and lengthy.

### Describe the solution you'd like.

I would like to have a docs section for Drifty with sections like - **Quickstart**, **Getting Started**, **Contributing**, **FAQ** and **Troubleshooting**.
- **Quickstart**: A brief introduction to the main features of Drifty, with a simple example to get users up and running in minutes.
- **Getting Started**: A more detailed guide on how to install, configure, and use Drifty, with explanations of the core components and options.
- **Contributing**: A section for developers who want to help make Drifty better, with instructions on how to set up a development environment, submit pull requests, and report bugs and new ideas.
- **FAQ**: A collection of common questions and answers about Drifty, covering topics such as compatibility, performance, security, and customization.
- **Troubleshooting**: This subsection helps to resolve common issues and errors that users may encounter while using Drifty and suggests possible solutions and workarounds.

The docs should be in **markdown files** and will be rendered by next js. You can read more about this [here](https://nextjs.org/docs/app/building-your-application/configuring/mdx).
> If you have any suggestions to add more suggestions, please let me know.

### Screenshots

The docs page can have a look like this -

![Image](https://github.com/SaptarshiSarkar12/Drifty/assets/105960032/a9534732-0cac-4bb3-b8d7-0979126d60ab)

> You can see the live version of the docs homepage [here](https://drifty-git-preview-docs-saptarshisarkar12.vercel.app/docs).

### Additional information
*Skills you need to solve this issue* - **NextJS**, **TailwindCSS**, [**MD / MDX in Nextjs**](https://nextjs.org/docs/app/building-your-application/configuring/mdx).
Those who wish to work on this issue, please leave a comment below and I will assign you with the task.

**Comment by sathvik-ng-07** [comment]
Hi SaptarshiSarkar12,

I'm interested in working on this issue. I've reviewed the problem and your proposed solution, and I'd like to contribute to improving the documentation for Drifty as described.

Here's what I plan to do:
- Create separate Markdown files for each documentation section (Quickstart, Getting Started, Contributing, FAQ, and Troubleshooting).
- Populate these files with clear and detailed content, including examples and code snippets where necessary.
- Update the main README file to link to the newly created documentation files.
- Test the changes locally to ensure they work as expected.
- Open a pull request (PR) once the changes are ready for review.

Please let me know if you have any specific requirements, guidelines, or additional information I should be aware of before I start working on this. I'm excited to contribute to the project and help improve its documentation.

Thank you!
Sathvik N G

**Comment by SaptarshiSarkar12** [comment]
Thank you @sathvik-ng-07 for your interest in this issue!
Yes, I agree with your plan. Looks good to me. :+1: 
You need to open the PR to the **master** branch (the default branch). **Please do not open any PR against other branches.**
I am assigning you with the task. You can start working on it.

**Comment by SaptarshiSarkar12** [comment]
@sathvik-ng-07 no longer wants to work on it. 
##### **If anyone else wants to work on this issue, please comment below and I will assign you with the task.**

**Comment by chauhansumitdev** [comment]
I would like to work on this issue.

**Comment by SaptarshiSarkar12** [comment]
> I would like to work on this issue.

Thank you for your interest in this issue!
I am assigning you with the task.

**Comment by markandey1414** [comment]
Hello @SaptarshiSarkar12 , I would like to work on this issue and resolve it.

**Comment by SaptarshiSarkar12** [comment]
@markandey1414 Yes, you can work on this issue. Assigning the task to you.
> You can discuss this issue in [our Discord server](https://discord.gg/DeT4jXPfkG).

**Comment by Neel-07** [comment]
hello @SaptarshiSarkar12, can you assign this issue to me i would like to work on this issue.

**Comment by SaptarshiSarkar12** [comment]
If @markandey1414 no longer wants to work on this issue, then, I can assign you (@Neel-07 ). Let's wait for his (@markandey1414 ) reply.

**Comment by markandey1414** [comment]
Yes @SaptarshiSarkar12 I'm working on it. Thanks for assigning me this.

**Comment by SaptarshiSarkar12** [comment]
Okay @markandey1414.

**Comment by markandey1414** [comment]
@SaptarshiSarkar12 permission to access the repo is denied, not able to issue pr

**Comment by SaptarshiSarkar12** [comment]
@markandey1414 Why? Can you please show the exact error?

**Comment by SaptarshiSarkar12** [comment]
@markandey1414 I think you have not made changes in your forked repo. I can't see your fork.
You should fork this repo, make changes there and then, open the Pull Request.

**Comment by markandey1414** [comment]
@SaptarshiSarkar12 I cloned the repo locally, made a branch, added a few files then tried to push it in the branch, but it says that permission is denied

**Comment by SaptarshiSarkar12** [comment]
You should clone your forked repo not this one. Otherwise, you will not be able to create a PR.

**Comment by SaptarshiSarkar12** [comment]
Read [here](https://jarv.is/notes/how-to-pull-request-fork-github/)

**Comment by SaptarshiSarkar12** [comment]
As @markandey1414 is not responding and has been inactive, I am removing the assignment from @markandey1414.
This issue is now open for all contributors and newcomers. So, if anyone wants to work on this issue, please comment below.

**Comment by hasnainmakada-99** [comment]
Hey @SaptarshiSarkar12 is anyone in queue for solving this issue, if no one's there then I would Like to try my hands on this issue

**Comment by hasnainmakada-99** [comment]
Alright then, Let's do this 🔥

**Comment by hasnainmakada-99** [comment]
Hey @SaptarshiSarkar12 Did I need to complete the docs section whole from the markdown itself or do I have to just create the individual files for it, like the getting started, FaQ and so on ?

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Yes, you need to create separate pages for each sections of the docs like **Getting Started**, **Contributing**, **FAQ**, etc.
@hasnainmakada-99 You can use either of the two - Your wish :smile:. I think MDx is better of the two, what do you say :thinking:?

**Comment by hasnainmakada-99** [comment]
> @hasnainmakada-99 Yes, you need to create separate pages for each sections of the docs like **Getting Started**, **Contributing**, **FAQ**, etc. @hasnainmakada-99 You can use either of the two - Your wish 😄. I think MDx is better of the two, what do you say 🤔?

Yeah sure @SaptarshiSarkar12 I will create the files as you said and coming from the flutter ecosystem, I am more familiar with Markdown and less with MDx although I know the basics of React but that would be not enough so can I proceed with General Markdown?

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Yes, you can proceed. 
> Please note that the data should be fetched from the Markdown - simply displaying/rendering the Markdown will not solve the issue.

**Comment by hasnainmakada-99** [comment]
Like which data @SaptarshiSarkar12, can you please provide more context to the note

**Comment by SaptarshiSarkar12** [comment]
You can check **[this](https://github.com/SaptarshiSarkar12/Drifty/tree/preview-docs/Website/app/docs)** branch to see how I have tried to implement a portion of the docs. It clearly shows how the work needs to be done. Suggestions to that approach are welcomed :smile:.
@hasnainmakada-99 Were you planning to do something like this?

**Comment by hasnainmakada-99** [comment]
Ohh I thought that I just have to provide the markdown files and the rest will be implemented by the dev team @SaptarshiSarkar12

**Comment by hasnainmakada-99** [comment]
I think that will be quite difficult for me to work on @SaptarshiSarkar12 as having no prior experience with next

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Okay. Then, try what you can do to solve the issue. No need to work in the way I have written the code.

**Comment by SaptarshiSarkar12** [comment]
Okay, then. No issue :+1:.

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 Did you face any new problems? How much is done? Please let me know.

**Comment by SaptarshiSarkar12** [comment]
Okay. No problem. You can take your time :smile:.

**Comment by SaptarshiSarkar12** [comment]
Most welcome :grin: @hasnainmakada-99!

**Comment by goody-dev** [comment]
@SaptarshiSarkar12  I'm a React developer, I'm new to contributing, and I'd like to look into this. It seems like the doc page isn't yet on the main website, right. But it's in the demo. Pls, am I to build the doc page and make a pull request or what exactly is expected?

**Comment by SaptarshiSarkar12** [comment]
@goody-dev Welcome to the Open-Source community :smile:! Thank you for your interest in this issue :pray:.
Sorry, currently this issue is being worked on by @hasnainmakada-99 via Pull Request #378. If he needs any help or you want to modify any of his changes, then, you are welcome to do so.

**Comment by hasnainmakada-99** [comment]
Hey @SaptarshiSarkar12 I'll resume the work by tonight if possible l, currently undergoing a medical treatment 😷

**Comment by SaptarshiSarkar12** [comment]
@hasnainmakada-99 It's okay. No problem. Have some rest :smile:, take care.

**Comment by goody-dev** [comment]
> @goody-dev Welcome to the Open-Source community :smile:! Thank you for your interest in this issue :pray:.
> Sorry, currently this issue is being worked on by @hasnainmakada-99 via Pull Request #378. If he needs any help or you want to modify any of his changes, then, you are welcome to do so.

Alright, no problem.

**Comment by SaptarshiSarkar12** [comment]
@goody-dev The one (@hasnainmakada-99) who has the issue assigned will not be able to create the feature to render the markdown files of docs. So, can you do that?

**Comment by goody-dev** [comment]
Yes, I would be glad to @SaptarshiSarkar12

**Comment by SaptarshiSarkar12** [comment]
Great 😃! Then, @hasnainmakada-99 will create the markdown files which contains the contents of docs pages. And you (@goody-dev) can create the feature that renders those docs pages contents. Is that okay for both of you?

**Comment by goody-dev** [comment]
Yes, @SaptarshiSarkar12. That's fine.

**Comment by SaptarshiSarkar12** [comment]
Okay. Thank you @goody-dev 👍.

**Comment by goody-dev** [comment]
Pls inform me when I can start working on it, @SaptarshiSarkar12. It seems like work is still going on the MD files.

**Comment by SaptarshiSarkar12** [comment]
@goody-dev You can start working on tha feature. For now, you can create test MD files which will get rendered. Later on, when @hasnainmakada-99's PR gets merged, I'll update your MD files with his.
Is that okay for you 🙂?

**Comment by goody-dev** [comment]
> @goody-dev You can start working on tha feature. For now, you can create test MD files which will get rendered. Later on, when @hasnainmakada-99's PR gets merged, I'll update your MD files with his.
> Is that okay for you 🙂?

That's alright.

**Comment by goody-dev** [comment]
@SaptarshiSarkar12, I have been unable to clone the repository to my local machine due to weak internet connection. I would get back to you if I have been able to clone it, and if anyone comes around before then and would to take on it they are free to.

**Comment by SaptarshiSarkar12** [comment]
Okay. It's fine. You can take your time :smile: @goody-dev.

**Comment by SaptarshiSarkar12** [comment]
@goody-dev Were you able to clone the repo?

**Comment by goody-dev** [comment]
No, I haven't been able to. And I have cloned another repo in between trying to.

**Comment by goody-dev** [comment]
No, I haven't been able to... And I have cloned another repo in between trying to.

**Comment by SaptarshiSarkar12** [comment]
Okay. If you are able to clone it later on, please inform me.

**Comment by SaptarshiSarkar12** [comment]
> No, I haven't been able to... And I have cloned another repo in between trying to.

@goody-dev Were you able to clone the repo successfully 😄?

**Comment by goody-dev** [comment]
Sorry, I won't be able to work on it at this time. you can assign others.

**Comment by SaptarshiSarkar12** [comment]
@goody-dev Okay. Thank you 👍🏻.

**Comment by abhilashreddysh** [comment]
@SaptarshiSarkar12 I can work on this. You can assign this issue to me.

**Comment by SaptarshiSarkar12** [comment]
@abhilashreddysh Thank you for your interest on this issue, Abhilash :smile:!
I am assigning the issue to you.
